src/query/expression/src/types/decimal.rs

```diff
@@ -65,17 +66,22 @@ pub struct DecimalSize {
     pub scale: u8,
 }
 
-pub trait Decimal: Sized {
+pub trait Decimal:
+    Sized + Default + Send + Sync + 'static + std::fmt::Display + std::fmt::Debug
+{
     fn one() -> Self;
     // 10**scale
     fn e(n: u32) -> Self;
 
+    fn max_of_max_precision() -> Self;
+
     fn min_for_precision(precision: u8) -> Self;
     fn max_for_precision(precision: u8) -> Self;
 
     fn from_float(value: f64) -> Self;
 
     fn try_downcast_column(column: &Column) -> Option<(Buffer<Self>, DecimalSize)>;
+    fn try_downcast_builder<'a>(builder: &'a mut ColumnBuilder) -> Option<&'a mut Vec<Self>>;
```

Comment:
I think try_downcast_builder also need return DecimalSize like try_downcast_column

Reply:
We just need to get the `Vec<_>` here. 
DecimalSize is already in DataType enum.